File: binding/include/libqaccessiblebridge.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qaccessiblebridge.html
pub const qaccessiblebridge = struct {

    /// New constructs a new QAccessibleBridge object.
    pub fn New() QtC.QAccessibleBridge {
        return qtc.QAccessibleBridge_new();
    }


    /// New2 constructs a new QAccessibleBridge object.
    pub fn New2(param1: ?*anyopaque) QtC.QAccessibleBridge {
        return qtc.QAccessibleBridge_new2(@ptrCast(param1));
    }


    pub fn SetRootObject(self: ?*anyopaque, rootObject: ?*anyopaque) void {
        qtc.QAccessibleBridge_SetRootObject(@ptrCast(self), @ptrCast(rootObject));
    }

    /// Allows for overriding the related default method
    pub fn OnSetRootObject(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QAccessibleBridge_OnSetRootObject(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetRootObject(self: ?*anyopaque, rootObject: ?*anyopaque) void {
        qtc.QAccessibleBridge_QBaseSetRootObject(@ptrCast(self), @ptrCast(rootObject));
    }

    pub fn NotifyAccessibilityUpdate(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QAccessibleBridge_NotifyAccessibilityUpdate(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnNotifyAccessibilityUpdate(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QAccessibleBridge_OnNotifyAccessibilityUpdate(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseNotifyAccessibilityUpdate(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QAccessibleBridge_QBaseNotifyAccessibilityUpdate(@ptrCast(self), @ptrCast(event));
    }

    pub fn OperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QAccessibleBridge_OperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QAccessibleBridge_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qaccessiblebridgeplugin.html
pub const qaccessiblebridgeplugin = struct {

    /// New constructs a new QAccessibleBridgePlugin object.
    pub fn New() QtC.QAccessibleBridgePlugin {
        return qtc.QAccessibleBridgePlugin_new();
    }


    /// New2 constructs a new QAccessibleBridgePlugin object.
    pub fn New2(parent: ?*anyopaque) QtC.QAccessibleBridgePlugin {
        return qtc.QAccessibleBridgePlugin_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAccessibleBridgePlugin_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qaccessiblebridgeplugin.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Create(self: ?*anyopaque, key: []const u8) QtC.QAccessibleBridge {
        const key_str = qtc.libqt_string{
    .len = key.len,
    .data = key.ptr,
};
return qtc.QAccessibleBridgePlugin_Create(@ptrCast(self), key_str);
    }

    /// Allows for overriding the related default method
    pub fn OnCreate(self: ?*anyopaque, callback: *const fn (?*anyopaque, [*:0]const u8) callconv(.c) QtC.QAccessibleBridge) void {
        qtc.QAccessibleBridgePlugin_OnCreate(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCreate(self: ?*anyopaque, key: []const u8) QtC.QAccessibleBridge {
        const key_str = qtc.libqt_string{
    .len = key.len,
    .data = key.ptr,
};
return qtc.QAccessibleBridgePlugin_QBaseCreate(@ptrCast(self), key_str);
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAccessibleBridgePlugin_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qaccessiblebridgeplugin.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAccessibleBridgePlugin_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qaccessiblebridgeplugin.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QAccessibleBridgePlugin_Delete(@ptrCast(self));
    }
};

File: binding/src/libqaccessiblebridge.cpp
#include <QAccessibleBridge>
#include <QAccessibleBridgePlugin>
#include <QAccessibleEvent>
#include <QAccessibleInterface>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qaccessiblebridge.h>
#include "libqaccessiblebridge.h"
#include "libqaccessiblebridge.hxx"

QAccessibleBridge* QAccessibleBridge_new() {
	 return new VirtualQAccessibleBridge();
}

QAccessibleBridge* QAccessibleBridge_new2(const QAccessibleBridge* param1) {
	 return new VirtualQAccessibleBridge(*param1);
}

void QAccessibleBridge_SetRootObject(QAccessibleBridge* self, QAccessibleInterface* rootObject) {
	self->setRootObject(rootObject);
}

void QAccessibleBridge_NotifyAccessibilityUpdate(QAccessibleBridge* self, QAccessibleEvent* event) {
	self->notifyAccessibilityUpdate(event);
}

void QAccessibleBridge_OperatorAssign(QAccessibleBridge* self, const QAccessibleBridge* param1) {
	self->operator=(*param1);
}

// Base class handler implementation
void QAccessibleBridge_QBaseSetRootObject(QAccessibleBridge* self, QAccessibleInterface* rootObject) {
	auto* vqaccessiblebridge = dynamic_cast<VirtualQAccessibleBridge*>(self);
	if (vqaccessiblebridge && vqaccessiblebridge->isVirtualQAccessibleBridge) {
vqaccessiblebridge->setQAccessibleBridge_SetRootObject_IsBase(true);
	vqaccessiblebridge->setRootObject(rootObject);
}
}

// Auxiliary method to allow providing re-implementation
void QAccessibleBridge_OnSetRootObject(QAccessibleBridge* self, intptr_t slot) {
	auto* vqaccessiblebridge = dynamic_cast<VirtualQAccessibleBridge*>(self);
	if (vqaccessiblebridge && vqaccessiblebridge->isVirtualQAccessibleBridge) {
vqaccessiblebridge->setQAccessibleBridge_SetRootObject_Callback(reinterpret_cast<VirtualQAccessibleBridge::QAccessibleBridge_SetRootObject_Callback>(slot));
}
}

// Base class handler implementation
void QAccessibleBridge_QBaseNotifyAccessibilityUpdate(QAccessibleBridge* self, QAccessibleEvent* event) {
	auto* vqaccessiblebridge = dynamic_cast<VirtualQAccessibleBridge*>(self);
	if (vqaccessiblebridge && vqaccessiblebridge->isVirtualQAccessibleBridge) {
vqaccessiblebridge->setQAccessibleBridge_NotifyAccessibilityUpdate_IsBase(true);
	vqaccessiblebridge->notifyAccessibilityUpdate(event);
}
}

// Auxiliary method to allow providing re-implementation
void QAccessibleBridge_OnNotifyAccessibilityUpdate(QAccessibleBridge* self, intptr_t slot) {
	auto* vqaccessiblebridge = dynamic_cast<VirtualQAccessibleBridge*>(self);
	if (vqaccessiblebridge && vqaccessiblebridge->isVirtualQAccessibleBridge) {
vqaccessiblebridge->setQAccessibleBridge_NotifyAccessibilityUpdate_Callback(reinterpret_cast<VirtualQAccessibleBridge::QAccessibleBridge_NotifyAccessibilityUpdate_Callback>(slot));
}
}

void QAccessibleBridge_Delete(QAccessibleBridge* self) {
    delete self;
}

QAccessibleBridgePlugin* QAccessibleBridgePlugin_new() {
	 return new VirtualQAccessibleBridgePlugin();
}

QAccessibleBridgePlugin* QAccessibleBridgePlugin_new2(QObject* parent) {
	 return new VirtualQAccessibleBridgePlugin(parent);
}

libqt_string QAccessibleBridgePlugin_Tr(const char* s) {
	QString _ret = QAccessibleBridgePlugin::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QAccessibleBridge* QAccessibleBridgePlugin_Create(QAccessibleBridgePlugin* self, const libqt_string key) {
	return self->create(QString::fromUtf8(key.data, key.len));
}

libqt_string QAccessibleBridgePlugin_Tr2(const char* s, const char* c) {
	QString _ret = QAccessibleBridgePlugin::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QAccessibleBridgePlugin_Tr3(const char* s, const char* c, int n) {
	QString _ret = QAccessibleBridgePlugin::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
QAccessibleBridge* QAccessibleBridgePlugin_QBaseCreate(QAccessibleBridgePlugin* self, const libqt_string key) {
	auto* vqaccessiblebridgeplugin = dynamic_cast<VirtualQAccessibleBridgePlugin*>(self);
	if (vqaccessiblebridgeplugin && vqaccessiblebridgeplugin->isVirtualQAccessibleBridgePlugin) {
vqaccessiblebridgeplugin->setQAccessibleBridgePlugin_Create_IsBase(true);
	return vqaccessiblebridgeplugin->create(QString::fromUtf8(key.data, key.len));
}
}

// Auxiliary method to allow providing re-implementation
void QAccessibleBridgePlugin_OnCreate(QAccessibleBridgePlugin* self, intptr_t slot) {
	auto* vqaccessiblebridgeplugin = dynamic_cast<VirtualQAccessibleBridgePlugin*>(self);
	if (vqaccessiblebridgeplugin && vqaccessiblebridgeplugin->isVirtualQAccessibleBridgePlugin) {
vqaccessiblebridgeplugin->setQAccessibleBridgePlugin_Create_Callback(reinterpret_cast<VirtualQAccessibleBridgePlugin::QAccessibleBridgePlugin_Create_Callback>(slot));
}
}

void QAccessibleBridgePlugin_Delete(QAccessibleBridgePlugin* self) {
    delete self;
}

